internal/fleet/package_resource.go

```diff
@@ -0,0 +1,120 @@
+package fleet
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/fleet"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+)
+
+func getPackageID(name, version string) string {
+	hash, _ := utils.StringToHash(name + version)
+
+	return *hash
+}
+
+func ResourcePackage() *schema.Resource {
+	packageSchema := map[string]*schema.Schema{
+		"name": {
+			Description: "The package name.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"version": {
+			Description: "The package version.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"force": {
+			Description: "Set to true to force the requested action.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+		},
+		"skip_destroy": {
```

Comment:
ubernit?

```suggestion
		"skip_uninstall": {
```

I think there's a better name here. You could argue that we're not skipping destroy (the resource _will_ be destroyed by TF, but the package won't be uninstalled). 

We have a `deletion_protection` attribute on the Elasticsearch index resource. That behaves differently to this (destroy will fail unless it's set to false), but I think it sets a precedent for this to be more API action focussed (e.g `skip_uninstall`) rather than TF action focussed.  

That's a lot of words for a minor naming discussion, I'm grasping for a clear argument :)

Reply:
We have `skip_destroy` on the agent policy, let's be consistent with that.

Reply:
I definitely agree that `skip_uninstall` would make more sense in this context, but yeah, I called it `skip_destroy` to be consistent with agent policies.